helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java

```diff
@@ -1033,6 +1033,14 @@ public void setAbnormalStateResolverMap(Map<String, String> resolverMap) {
     return idealStateRuleMap;
   }
 
+  /**
+   * Create an instance of {@link ClusterTopologyConfig} based on cluster config.
+   * @return an instance of {@link ClusterTopologyConfig}.
+   */
+  public ClusterTopologyConfig getClusterTopologyConfig() {
+    return ClusterTopologyConfig.fromClusterConfig(this);
```

Comment:
Seems a bit odd to me... Is there a reason why ClusterConfig has a method to construct a ClusterTopologyConfig out of itself? If ClusterTopologyConfig needs to be constructed, the caller should just directly call `ClusterTopologyConfig.fromClusterConfig(clusterConfig)`, instead of `clusterConfig.getClusterTopologyConfig()`; this method doesn't seem that necessary to me.

Reply:
Good question, indeed it's not the most beautiful way... I choose this approach to avoid over expose this class, right now `ClusterTopologyConfig.fromClusterConfig` is package private.
In my view, this class is like a view of `ClusterConfig`, a container associated with ClusterConfig, and how we construct this is impl details and shouldn't be exposed outside package. I didn't write the logic directly in ClusterConfig because this class is already too long.
This is open to discussion, let me know how you feel about it. Thanks!

Reply:
I believe ClusterTopologyConfig is moved out of the class to make it independent. It makes more sense to me that ClusterTopologyConfig class is called to create itself.

Reply:
Fair enough, updated. Thanks.